models/src/test/scala/coop/rchain/models/blockImplicits.scala

```diff
@@ -35,15 +38,60 @@ object blockImplicits {
 
   implicit val arbitraryJustification: Arbitrary[Justification] = Arbitrary(justificationGen)
 
+  val defaultSec = PrivateKey(
+    Base16.unsafeDecode("a68a6e6cca30f81bd24a719f3145d20e8424bd7b396309b0708a16c7d8000b76")
+  )
+
+  private def fill(
+      deployData: DeployData
+  )(deployer: PublicKey, sig: Array[Byte], sigAlgorithm: String): DeployData =
+    deployData
+      .withDeployer(ByteString.copyFrom(deployer.bytes))
+      .withSig(ByteString.copyFrom(sig))
+      .withSigAlgorithm(sigAlgorithm)
+
+  private def clear(deployData: DeployData): DeployData =
+    fill(deployData)(PublicKey(Array.empty[Byte]), Array.empty[Byte], "")
+
+  private def sign(sec: PrivateKey, deployData: DeployData): DeployData = {
+    val toSign    = clear(deployData).toByteString.toByteArray
+    val hash      = Blake2b256.hash(toSign)
+    val signature = Secp256k1.sign(hash, sec)
+
+    fill(deployData)(Secp256k1.toPublic(sec), signature, Secp256k1.name)
+  }
+
+  val processedDeployGen: Gen[ProcessedDeploy] =
+    for {
+      term      <- listOfN(32, Gen.alphaNumChar).map(_.mkString)
+      timestamp <- arbitrary[Long]
+      deployData = DeployData(
+        deployer = ByteString.copyFrom(Secp256k1.toPublic(defaultSec).bytes),
+        timestamp = timestamp,
+        term = term,
+        phloLimit = 90000,
+        phloPrice = 1L
+      )
+      signed = sign(defaultSec, deployData)
+    } yield ProcessedDeploy(
+      deploy = Some(signed)
+    )
```

Comment:
quite a lot of this code seems to be duplicated from other modules. Rather than copying code to here, we should add any implicits needed for generating random BlockMessages in the module BlockMessage belongs - casper.

Reply:
Also/alternatively: I don't think any of your tests rely on the deploys being signed or having proper phlo limit. Stick to totally random data maybe?

Reply:
> we should add any implicits needed for generating random BlockMessages in the module BlockMessage belongs - casper.

Moving the implicits for generating random BlockMessages to casper module would mean that we won't be able to use them from blockstorage module (or we will have to make it depend on casper which we can't do because it will be a cyclic dependency).

> Stick to totally random data maybe?

You are right, using random data is probably better than copy-pasting code.